open-craft/edx-theme
# Rewrite course_about.html to use new blocks

This, together with https://github.com/open-craft/edx-platform/pull/96, makes the Pearson theme continue working with no design changes, but simplifies the HTML (in about_course.html only) and makes it easier to maintain.
It uses a new way of creating themes: by inheriting from the base theme and redefining only the blocks that change. For this it requires the code in https://github.com/edx/edx-platform/pull/16856/commits and the named blocks in https://github.com/open-craft/edx-platform/pull/96
It also add explanations about what was changed in the Pearson theme, which should help when we migrate this theme to a newer Open edX release.

**JIRA tickets**: None
**Discussions**: None
**Dependencies**: None
**Screenshots**: None
**Sandbox URL**: None
**Merge deadline**: None

**Testing instructions**:

1. Either:
    - Deploy in a server which includes both https://github.com/edx/edx-platform/pull/16856 and  https://github.com/open-craft/edx-platform/pull/96 on top of the pearson branch  https://github.com/open-craft/edx-platform/tree/opencraft-release/ginkgo.1-pearson (it's based on ginkgo.1; works in vagrant and docker)
    - or log in to the one in https://console.opencraft.com/instance/2760/edx-appserver/4118/ , where I already did those changes
2. Create or change the introduction course to add  a video so that we test all components. (Or if using Ocim's appserver, locate a course with video, e.g. the one about digital marketing)
3. Visit http://localhost:18000/courses/course-v1:edX+DemoX+Demo_Course/about (or https://courses.pearsonx.com/courses/course-v1:WhartonX+DM-17-18+2017-18_T1/about if you're using the Ocim appserver), you're still seeing the original page (this PR is unapplied yet). Save the HTML to a file
4. Activate the theme in this PR (not needed if using the Ocim appserver because it's already enabled) 
5. Go to the same page in the theme-enabled appserver and save the HTML to another file
6. Compare both HTMLs, they should be +the same+ equivalent. That is, the original one (no blocks) and the new one (which only defines specific blocks) should produce a complete page. Only two things changed:
    - A lone `<div>` around `<ol class="important-dates">` was removed; this doesn't modify the design (check it visually)
    - The JS order was changed. To check that it's working, try http://localhost:18000/courses/course-v1:edX+DemoX+Demo_Course/about#contacted too (it shows a new blue box)
7. Compare visually with https://courses.pearsonx.com/courses/course-v1:ThunderbirdX+IBM-17-18+2018_T2/about or with the course you chose
8. Test that the page is still responsive in small screen
9. Try the video part by clicking it
10. Read the explanations about what was changed in the Pearson branch. Compare them with this diff (base ginkgo.1 templates → our Pearson templates): https://github.com/open-craft/edx-theme/compare/pearsonx-upstream-templates...open-craft:pearsonx (only for course_about.html)

**Reviewers**

**Author concerns**: None
**Settings**: None
